### DISCUSSION TITLE
Execution order of methods annotated with @AfterScenario

### DISCUSSION THREAD

**Discussion by knapik-m** [getgauge/gauge#2100] [General]
Hello,

I have scenario with two @AfterScenario methods - one with tag and second one without tags. According to documentation if there is multiple methods with this annotation, execution order is as follows:
- Hooks which are not filtered by tags.
- Hooks filtered by tags.

I am using gauge-java and in my scenario a method without tags is executed as second one. Is this an expected behaviour? If so, how can I execute @AfterScenario methods in defined order? 

Thank you.

**Comment by sriv** [comment]
Hi @knapik-m 

Could you please point out where you observed this behaviour documented? (Sorry, but my search yielded nothing)

This behaviour is controlled by the runner, and looking at gauge-java, this rule is not in place! From memory, I believe gauge-dotnet exhibits this behaviour.

**Reply by knapik-m** [reply]
Hi @sriv 

It is documented in gauge-java repository in @AfterScenario class:

https://github.com/getgauge/gauge-java/blob/master/src/main/java/com/thoughtworks/gauge/AfterScenario.java
https://www.javadoc.io/doc/com.thoughtworks.gauge/gauge-java/latest/com/thoughtworks/gauge/AfterScenario.html

And yes, I also checked the implementation in gauge-java repository and saw that there is no guaranteed execution order of methods annotated with AfterScenario.
